Upgrade fidesops base image from Debian Buster to Bullseye

fidesops is currently based on python:3.x.y-slim-buster. In August 2022 Debian Buster (10) is [scheduled](https://wiki.debian.org/DebianReleases#Production_Releases) to move from official (oldstable) support to [LTS](https://wiki.debian.org/LTS) community support.

The fidesops base image should be changed to python:3.x.y-slim-bullseye to ensure that timely security and version updates are applied to the application image.

The `ipython` one is not a big deal. I talk to the team and everyone said they don't use it so we should be able to remove it all together. The bigger issue is mssql also has an issue. The drivers can't be installed in bullseye on M1 Macs. There is [an issue](https://github.com/MicrosoftDocs/sql-docs/issues/6494) about this, but Microsoft closed it without a solution so I'm not sure if they plan to fix it.

```console
 > [backend  8/13] RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then apt-get -y --no-install-recommends install     unixodbc-dev     msodbcsql17     mssql-tools     && apt-get clean     && rm -rf /var/lib/apt/lists/* ; fi:
#12 0.300 Reading package lists...
#12 2.766 Building dependency tree...
#12 3.240 Reading state information...
#12 3.341 Some packages could not be installed. This may mean that you have
#12 3.341 requested an impossible situation or if you are using the unstable
#12 3.341 distribution that some required packages have not yet been created
#12 3.341 or been moved out of Incoming.
#12 3.341 The following information may help to resolve the situation:
#12 3.341 
#12 3.341 The following packages have unmet dependencies:
#12 3.576  libodbc1 : PreDepends: multiarch-support but it is not installable
#12 3.577  odbcinst1debian2 : PreDepends: multiarch-support but it is not installable
#12 3.601 E: Unable to correct problems, you have held broken packages.
------
executor failed running [/bin/sh -c if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then apt-get -y --no-install-recommends install     unixodbc-dev     msodbcsql17     mssql-tools     && apt-get clean     && rm -rf /var/lib/apt/lists/* ; fi]: exit code: 100
make: *** [docker-build] Error 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,7 +25,10 @@ The types of changes are:
 * Access and erasure support for Adobe Campaign [#905](https://github.com/ethyca/fidesops/pull/905)
 
 ### Changed
+
 * Users should be able to click on the full field of a dropdown-type filter to open up the dropdown [#747](https://github.com/ethyca/fidesops/pull/903)
+* Updated the python docker base image from slim-buster to slim-bullseye [928](https://github.com/ethyca/fidesops/pull/928)
+* Removed ipython from the docker install [928](https://github.com/ethyca/fidesops/pull/928)
 
 ### Breaking Changes
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -13,7 +13,7 @@ RUN npm run export
 #############
 ## Backend ##
 #############
-FROM --platform=linux/amd64 python:3.9.13-slim-buster as backend
+FROM --platform=linux/amd64 python:3.9.13-slim-bullseye as backend
 
 
 # Install auxiliary software
@@ -21,7 +21,6 @@ RUN apt-get update && \
     apt-get install -y --no-install-recommends \
     git \
     make \
-    ipython \
     vim \
     curl \
     g++ \
@@ -39,7 +38,7 @@ RUN echo "ENVIRONMENT VAR:  SKIP_MSSQL_INSTALLATION $SKIP_MSSQL_INSTALLATION"
 # https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver15
 RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then apt-get install -y --no-install-recommends apt-transport-https && apt-get clean && rm -rf /var/lib/apt/lists/* ; fi
 RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - ; fi
-RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then curl https://packages.microsoft.com/config/debian/10/prod.list | tee /etc/apt/sources.list.d/msprod.list ; fi
+RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then curl https://packages.microsoft.com/config/debian/11/prod.list | tee /etc/apt/sources.list.d/msprod.list ; fi
 RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then apt-get update ; fi
 ENV ACCEPT_EULA=y DEBIAN_FRONTEND=noninteractive
 RUN if [ "$SKIP_MSSQL_INSTALLATION" != "true" ] ; then apt-get -y --no-install-recommends install \

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -1,6 +1,5 @@
 black==22.6.0
 Faker==13.15.0
-ipython
 isort
 mypy==0.971
 packaging==21.3

